GSA-TTS/FAC
# Updated code to prevent it from sending extra empty fields to the database

## Description
This pull request aims to remove unnecessary fields from the general information section. Recently, the schema and validation rules for this section were updated to include mandatory address fields for auditor country information. Specifically, U.S. address fields were required when the auditor's country was listed as the USA, and international address fields were required for other countries. However, the logic change inadvertently made all address fields mandatory, which interfered with our data migration algorithm by causing failures due to irrelevant missing address fields for the records being migrated. This PR addresses that issue. The proposed solution includes updating the schema to require address fields only when applicable, removing unnecessary empty address fields upon form submission, and modifying the validation logic to disregard these fields in reports that are already underway, ensuring that submissions are not blocked as this code change is released.

## How to test

1. Launch the stack from the main banch and create an audit report. 
2. Switch the auditor country between USA and non-USA and notice the extra empty fields in database sac object (general information section).
3. Checkout this PR branch and repeat step 2. Observe that the extra address fields are properly removed each time you switch back and forth between USA and non-USA auditor country.
4. To ensure in-progress reports are not affected:
            Switch to the main branch again and create a new report or use an existing one (if any). 
             When ready to intiate final validation for certification, switch to this PR branch and initiate that final validation.
             Note that the report validate properly despite having extra address fields.

## PR checklist: submitters


## PR checklist: reviewers


The larger the PR, the stricter we should be about these points.
